contracts/DutchExchange.sol

```diff
@@ -1070,6 +1074,38 @@ contract DutchExchange is DxUpgrade, TokenWhitelist, EthOracle {
         auctionIndex = latestAuctionIndices[token1][token2];
     }
 
+    function setClearingTime(
```

Comment:
The function is always closing using the start time and the closing price. This is good for the close theoreticals, but wouldn't we save gas in the normal case, were we close the auction because of a trade?

Reply:
In theory yes, although the gas savings are imo negligible - it's just calculation, no storage i/o. In practice in `clearAuction` we don't know in which way it's been cleared, so we would have to rework the entire clearing mechanism. This would cause compatibility issues and security risk.

Reply:
okay, if it's simpler, and is gas increase negligible, okay :)